cooltools/api/eigdecomp.py

```diff
@@ -499,34 +472,44 @@ def eigs_trans(
     A = clr.matrix(balance=clr_weight_name)[lo:hi, lo:hi]
     bins = clr.bins()[lo:hi]
 
+    phasing_vector = None
     if phasing_track is not None:
         phasing_track = align_track_with_cooler(
             phasing_track,
             clr,
             view_df=view_df,
             clr_weight_name=clr_weight_name,
             mask_clr_bad_bins=True,
-            drop_track_na=True # this adds check for chromosomes that have all missing values
+            drop_track_na=True  # this adds check for chromosomes that have all missing values
         )
-        phasing_track_values = phasing_track["value"].values[lo:hi]
-    else:
-        phasing_track_values = None
-
-    eigvals, eigvecs = trans_eig(
-        A,
-        partition,
-        n_eigs=n_eigs,
-        phasing_track=phasing_track_values,
-        sort_metric=sort_metric,
-        **kwargs,
-    )
+        phasing_vector = phasing_track["value"].values[lo:hi]
+
+    OE = _obsexp_trans_dense(A, partition, perc_top, perc_bottom)
+
+    eigvecs, eigvals = numutils.get_eig(OE, n_eigs, mask_zero_rows=True)
+    eigvecs /= np.sqrt(np.nansum(eigvecs ** 2, axis=1))[:, None]
+    eigvecs *= np.sqrt(np.abs(eigvals))[:, None]
+
+    if phasing_vector is not None:
+        corrs = _correlate_with_eigs(
+            eigvecs, phasing_vector, corr_metric
+        )
+        # Flip signs of eigvecs deterministically
+        for i in range(len(corrs)):
+            eigvecs[i] = np.sign(corrs[i]) * eigvecs[i]
+        # Re-rank eigvecs by descending correlation to phasing track
+        if reorder:
+            idx = np.argsort(-np.abs(corrs))
+            eigvals = eigvals[idx]
+            eigvecs[idx] = eigvecs[idx]
 
     eigvec_table = bins.copy()
```

Comment:
can we align this part - the fill in part - with the `eigs_cis` function - to facilitate maintenance ...

```python
eigvec_columns = [f"E{i + 1}" for i in range(n_eigs)]

eigvec_table = bins.reindex(columns=bins.columns+eigvec_columns)
eigvec_table[eigvec_columns] = eigvecs.T
```